### ISSUE TITLE
AirPlay device commences playback at last played volume not current volume

### ISSUE THREAD

**Issue by OzGav** [music-assistant/support#3330]
### Discussed in https://github.com/orgs/music-assistant/discussions/3322

<div type='discussions-op-text'>

<sup>Originally posted by **frank2604** December 20, 2024</sup>
I can reproduce the behavior in the MA UI as well:

1. Player plays at volume 50:
<img width="498" alt="2024-12-20_11-20-52" src="https://github.com/user-attachments/assets/85ad9656-2645-4721-ae46-843b51cb16a1" />

2. Turn off player:
<img width="504" alt="2024-12-20_11-21-17" src="https://github.com/user-attachments/assets/e0a9f758-825d-47cc-986e-3043164e8b31" />

3. Turn on player:
<img width="492" alt="2024-12-20_11-23-22" src="https://github.com/user-attachments/assets/d8f419ca-6b89-419a-8394-aa9f0c087471" />

4. Set volume to 10:
<img width="494" alt="2024-12-20_11-24-55" src="https://github.com/user-attachments/assets/5120d3c3-7763-4d7e-a136-4ff8d7448e0d" />

5. Start playback:
<img width="495" alt="2024-12-20_11-25-16" src="https://github.com/user-attachments/assets/30561b38-2b0f-4c17-9fce-1a897d2ff8f1" />

6. It will play at the last used volume of 50:
<img width="497" alt="2024-12-20_11-25-38" src="https://github.com/user-attachments/assets/28f3147b-3370-4715-b351-a5c9ad2c288f" />

**Comment by marcelveldt** [comment]
I can't reproduce this on 2.4.0rc4 - this can just be a player issue because some players ignore the volume level at start of playback and just overrule it with whatever they have set.

Anyways, I tried 3 different airplay devices and it works as expected.

**Comment by OzGav** [comment]
@ frank2604 please try with rc4 and see what you find.

**Comment by OzGav** [comment]
We will close this soon assuming fixex

**Comment by frank2604** [comment]
I would like to try it. How can I install 2.4.0rc4? Im am running MA via HA Add-on. Can#t find any option to switch to beta.

**Comment by OzGav** [comment]
Just wait a couple of days and the rc will be promoted to stable. I f you do want to try it out look in the addon store and you will see the beta. https://www.music-assistant.io/release/#running-parallel-server-versions

**Comment by OzGav** [comment]
@frank2604  2.4.0 has been released.

**Comment by OzGav** [comment]
This will be closed soon assuming fixed

**Comment by OzGav** [comment]
@marcelveldt Actually I can reproduce this with my Yamaha Amplifier doing the steps in the first post

**Comment by marcelveldt** [comment]
Most likely this is an issue then with receivers or specific players, ignoring the startup volume.
If this is really troublesome I could try to send an extra volume call at start of stream.

**Comment by marcelveldt** [comment]
Well, from the perspective of the receiver it actually makes sense to ignore the volume at playback start...
Imagine you're watching TV at volume 10 and then start music playback from MA but you were listening to MA music on that receiver yesterday at volume 30. Do you actually want the receiver to start the music blasting at volume 30 or at volume 10 ?

Within the Airplay protocol we have a way to send the initial volume with the call to start the music but I guess some/many manufacturers decided to ignore that for the reason I mentioned above. MA is not the only source and it is better to just keep the current volume than some randomly stored previous volume. 

Remember that only when playback is started, the player will start sending its current volume back to MA.

We could make it an option to enforce sending the volume at playback start ?

**Comment by OzGav** [comment]
That sounds like an easy solution so people can choose.

**Comment by frank2604** [comment]
@marcelveldt 
You could just argue the other way around: If you have been listening to loud music during the day and want to listen to some quiet background music at your desk at night, you want to set a quiet level before you start playing so as not to disturb everyone in the house.

My use case is that I want the music to fade in when I enter certain rooms. Before playing, I want to set the volume to 1, start playing and then let it run up to 30 in a loop.

So an option to enforce sending the volume at playback start is a great idea.

**Comment by OzGav** [comment]
@frank2604 please confirm this has been fixed in 2.5.0b18

**Comment by frank2604** [comment]
@OzGav Thanks. I will test it, wenn 2.5.0 is out.

**Comment by OzGav** [comment]
@frank2604 2.5.0 has been released. Please confirm this is fixed

**Comment by frank2604** [comment]
@OzGav 

Unfortunately, nothing has changed in this behavior.

1. Playback with MA at 30% on the MusicCast player 
2. Pause playback. 
3. Set the player in Standby on the hardware (Power off is not available for this player in MA).
4. Set the volume to 10% in MA
5. Start Playing in MA again.
6. The volume control in MA remains at 10%, but the player plays at 30%.

I think this is a problem specifically with Yamaha MusicCast. This leads us to think that a player provider for MC would be great.

PS: Now MA doesn't turn on the MC player 9 times out of 10 when it was previously set to standby. I haven't been able to figure out when and after what wait time it works and when it doesn't. It seems to be completely random. Should I open a new case for this?

**Comment by OzGav** [comment]
On your last point have you mapped a HA power control to the MA player?

Now on your first point, you are now talking about MusicCast when the OP was about AirPlay. For this report we just need to confirm it is fixed when using an AirPlay device

**Comment by frank2604** [comment]
> On your last point have you mapped a HA power control to the MA player?

Do you mean the Fake Power Control in the Player Controls? Yes, I have. Or is there anything else I did not know so far?

> Now on your first point, you are now talking about MusicCast when the OP was about AirPlay. For this report we just need to confirm it is fixed when using an AirPlay device

The hardware are MusicCast devices. They are established in MA via AirPlay Player Provider. So in my understanding it is an AirPlay device.

**Comment by OzGav** [comment]
You should map the HA entity power control https://www.music-assistant.io/ha-plugin/

Please just use AirPlay when you are talking about this so we know what provider you are talking about. I will try and recreate this again.

**Comment by OzGav** [comment]
I am no longer able to recreate this. Have you tried changing this setting? 

![Image](https://github.com/user-attachments/assets/6d17e74a-6d57-4839-aeb2-fca4d65d9993)

**Comment by frank2604** [comment]
I tried it now, but it doesn't change anything.

As I mentioned before: I guess this is a problem specifically with the Yamaha MusicCast AirPlay implementation. A player provider for MC would be great.

**Comment by OzGav** [comment]
Yes. It is a feature request so make sure you have upvoted it https://github.com/orgs/music-assistant/discussions/409

Does the mapped power control fix that problem?

As for the main problem will have to wait and see if Marcel has any other ideas.

**Comment by frank2604** [comment]
> Does the mapped power control fix that problem?

Do you mean the Fake Power Control in the Player Controls? No that hasn't solved the problem.

**Comment by OzGav** [comment]
No not the fake power control. Look at the link I provided above. You need to map a HA entity to the MA player. That is, you probably have the device discovered by HA. HA will have created a switch entity. You can map that to MA.

**Comment by frank2604** [comment]
Understand. Now I have mapped the player entity (there is no specific power entity), but that doesn't change anything.

<img width="777" alt="Image" src="https://github.com/user-attachments/assets/64db5940-6348-428d-893d-e69d8a3ac765" />

**Comment by OzGav** [comment]
This is just to fix "Now MA doesn't turn on the MC player 9 times out of 10 when it was previously set to standby."

**Comment by OzGav** [comment]
MusicCast has been added to 2.6.0b6 Please try that instead of AirPlay

**Comment by marcelveldt** [comment]
What is the status of this ticket ? I cant follow any of it as its talking about 3 different topics at once.

**Comment by OzGav** [comment]
This appears to be a device issue as I could not recreate it with my Yamaha amplifier. The ignore volume reports settings has been reported to solve similar issues in other threads so that appears to be working. The MusicCast provider has been released which should now be used. I think this can be closed.
